lib/kafka/async_producer.rb

```diff
@@ -128,18 +120,36 @@ def deliver_messages
     # @return [nil]
     def shutdown
       @queue << [:shutdown, nil]
-      @worker_thread.join
+      @worker_thread && @worker_thread.join
 
       nil
     end
 
+    private
+
+    def ensure_threads_running!
+      @worker_thread = nil unless @worker_thread && @worker_thread.alive?
+      @worker_thread ||= start_thread { @worker.run }
+
+      @timer_thread = nil unless @timer_thread && @timer_thread.alive?
+      @timer_thread ||= start_thread { @timer.run }
+    end
+
+    def start_thread(&block)
+      thread = Thread.new(&block)
+      thread.abort_on_exception = true
+      thread
+    end
+
     class Timer
       def initialize(interval:, queue:)
         @queue = queue
         @interval = interval
       end
 
       def run
+        Thread.stop if @interval.zero?
```

Comment:
My understanding is that this will put the thread in a permanent sleep state:

```
irb(main):001:0> t = Thread.new { Thread.stop; puts "hello" }
=> #<Thread:0x007ffb4984ff98@(irb):1 run>
irb(main):002:0> t.alive?
=> true
```